WalletWasabi/Blockchain/BlockFilters/IndexBuilderService.cs

```diff
@@ -150,116 +116,75 @@ public void Synchronize()
 									syncInfo = await GetSyncInfoAsync();
 								}
 
-								uint heightToRequest = StartingHeight;
+								uint currentHeight = 0;
 								uint256 currentHash = null;
 								using (await IndexLock.LockAsync())
 								{
 									if (Index.Count != 0)
 									{
 										var lastIndex = Index[^1];
-										heightToRequest = lastIndex.Header.Height + 1;
+										currentHeight = lastIndex.Header.Height;
 										currentHash = lastIndex.Header.BlockHash;
 									}
+									else
+									{
+										currentHash = StartingHeight == 0
+											? uint256.Zero
+											: await RpcClient.GetBlockHashAsync((int)StartingHeight - 1);
+										currentHeight = StartingHeight - 1;
```

Comment:
For regtest the `StartingHeight` is 0. Given this is a `ulong`, in this line the `currentHeight` results in a huge number. This makes all the subsequent comparisons `false`.

Reply:
True but this is an edge case only to effect the first block in regtest.

Reply:
I am Mr. Case, Edge Case.